### PR TITLE
Prod 1041/dictionary data uses

### PR DESCRIPTION
**Pull request by jpople**
### Description Of Changes

Adds UI for populating data uses on a system using the dictionary.

Note: dictionary suggestions require Fideslang 2.0 to be guaranteed to work, some keys have changed and the backend won't accept them if you're using the old ones.  See [here](https://ethyca.slack.com/archives/C04S7K0T2V7/p1694026429700819) for more context.

Screenshot 2023-09-06 at 5 44 12 PM
Screenshot 2023-09-06 at 5 44 27 PM
Screenshot 2023-09-06 at 5 44 45 PM
Screenshot 2023-09-06 at 5 44 50 PM

### Steps to Confirm

* [ ] Add a new system
* [ ] Save new system
* [ ] Go to "data uses" tab-- dictionary button should be greyed out, and on hover, a tooltip should show
* [ ] Assign a vendor to the system in "system information" tab
* [ ] Generate dictionary suggestions with "generate data uses automatically" button
* [ ] Accept suggestions

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx [hunk 1]
```diff
@@ -182,6 +218,23 @@ const PrivacyDeclarationFormTab = ({
     setCurrentDeclaration(declarationToEdit);
   };
 
+  const handleOpenDictModal = () => {
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx:221]: nit: chakraui has a build in modal handler that automatically creates these functions: https://v1.chakra-ui.com/docs/styled-system/utility-hooks/use-disclosure

using that instead might be nicer

File: clients/admin-ui/src/features/system/dictionary-data-uses/EmptyTableState.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  handleDictSuggestion,
+  vendorSelected,
+}: Props) => {
+  const dictDisabledTooltip = useMemo(
+    () =>
+      "You will need to select a vendor for this system before you can use the Fides dictionary. You can do this on System Information tab above.",
+    []
+  );
+
+  console.log(vendorSelected);
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/dictionary-data-uses/EmptyTableState.tsx:30]:
```suggestion

```

File: clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx [hunk 3]
```diff
@@ -51,10 +55,19 @@ const PrivacyDeclarationFormTab = ({
 
   const [updateSystemMutationTrigger] = useUpdateSystemMutation();
   const [showForm, setShowForm] = useState(false);
+  // const [showDictionaryModal, setShowDictionaryModal] = useState(false);
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx:58]:
```suggestion

```

File: clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx [hunk 4]
```diff
@@ -182,6 +225,23 @@ const PrivacyDeclarationFormTab = ({
     setCurrentDeclaration(declarationToEdit);
   };
 
+  // const handleOpenDictModal = () => {
+  //   setShowDictionaryModal(true);
+  // };
+
+  // const handleCloseDictModal = () => {
+  //   setShowDictionaryModal(false);
+  // };
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationFormTab.tsx:234]:
```suggestion

```

File: clients/admin-ui/src/features/system/dictionary-data-uses/DataUseCheckboxTable.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    return declaration.data_use;
+  };
+
+  const allChecked = dictDataUses.length === checked.length;
+
+  return (
+    <Table variant="unstyled" size="sm" border="1px" borderColor="gray.200">
+      <Thead border="1px" borderColor="gray.200" backgroundColor="gray.50">
+        <Tr sx={{ "border-collapse": "separate" }}>
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/dictionary-data-uses/DataUseCheckboxTable.tsx:66]:
```suggestion
        <Tr borderCollapse="separate">
```

I got this error in the console. I think this is what it's asking for

Screenshot from 2023-09-07 13-41-41

Comment (TheAndrewJackson) [reply clients/admin-ui/src/features/system/dictionary-data-uses/DataUseCheckboxTable.tsx:66]: if that doesn't work then `styte={{"border-collapse": "separate"}}` might work as well

Comment (jpople) [reply clients/admin-ui/src/features/system/dictionary-data-uses/DataUseCheckboxTable.tsx:66]: Oops, yeah, I left that in from when I was testing some stuff before, it's not doing anything.

**Review by TheAndrewJackson — APPROVED** [review]
:boom:

**Pull request merged**